### PR TITLE
Fix speed differential for retail cap ships

### PR DESCRIPTION
**Pull request by wookieejedi**
As astutely determined by @Baezon `the speed differential for retail cap ships is caused by 'ships with no shields can manage ETS', being part of the bugfixes with target version`. This PR adds two flags per Discord suggestion, one for all ships and one for only fighters/bombers.

### CODE REVIEW

**Review by Baezon — APPROVED** [review]

File: code/ai/ai_profiles.cpp [hunk 1]
```diff
@@ -525,7 +525,9 @@ void parse_ai_profiles_tbl(const char *filename)
 
 				set_flag(profile, "$respect ship axial turnrate differences:", AI::Profile_Flags::Use_axial_turnrate_differences);
 
-				set_flag(profile, "$ships with no shields can manage ETS:", AI::Profile_Flags::nonshielded_ships_can_manage_ets);
+				set_flag(profile, "$ships with no shields can manage ETS:", AI::Profile_Flags::all_nonshielded_ships_can_manage_ets);
```
Comment (Goober5000) [inline code/ai/ai_profiles.cpp:528]: I would make the text explicit, such as `$any ship with no shields can manage ETS:`.  Keep the old string too, for compatibility, but add a deprecation warning.

File: code/ai/ai_profiles.cpp [hunk 2]
```diff
@@ -525,7 +525,9 @@ void parse_ai_profiles_tbl(const char *filename)
...
+
+				set_flag(profile, "fighters/bombers with no shields can manage ETS:", AI::Profile_Flags::fightercraft_nonshielded_ships_can_manage_ets);
```
Comment (Goober5000) [inline code/ai/ai_profiles.cpp:530]: Add a `$`.

File: code/hud/hudets.cpp [hunk 3]
```diff
@@ -212,10 +212,14 @@ void ai_manage_ets(object* obj)
...
+	if (!ship_info_p->max_speed || !ship_info_p->max_weapon_reserve) {
+		return;
+	}
+
 	// also check if the ship has no shields and if the AI is allowed to manage weapons and engines --wookieejedi
-	if ( !ship_info_p->max_speed || !ship_info_p->max_weapon_reserve || 
-		(!ship_p->ship_max_shield_strength && !aip->ai_profile_flags[AI::Profile_Flags::nonshielded_ships_can_manage_ets]) ) {
-			return;
+	if (!ship_p->ship_max_shield_strength && !(aip->ai_profile_flags[AI::Profile_Flags::all_nonshielded_ships_can_manage_ets] || 
+		(ship_info_p->is_fighter_bomber() && aip->ai_profile_flags[AI::Profile_Flags::fightercraft_nonshielded_ships_can_manage_ets])) ) {
```
Comment (Baezon) [inline code/hud/hudets.cpp:221]: Actually, the logic here is a bit confusing, could we get some more parentheses to make it more clear?

**Review by Baezon — CHANGES_REQUESTED** [review]

**Review by Goober5000 — CHANGES_REQUESTED** [review]
Couple things, otherwise this looks good.

File: code/ai/ai_profiles.cpp (hunk 1 above)
Comment (wookieejedi) [reply code/ai/ai_profiles.cpp]: Good idea, committed!

**Review by Goober5000 — APPROVED** [review]

File: code/hud/hudets.cpp (hunk 3 above)
Comment (wookieejedi) [reply code/hud/hudets.cpp]: Definitely

**Review by Baezon — APPROVED** [review]

**Pull request merged**